### DISCUSSION TITLE
numberize

### DISCUSSION THREAD

**Discussion by bahadzie** [epiverse-trace/epiverse-trace.github.io#92] [Show and tell]
_numberize_ is an R package to convert numbers written as English, French or Spanish words from `"zero"` to `"nine hundred and ninety nine trillion, nine hundred and ninety nine billion, nine hundred and ninety nine million, nine hundred and ninety nine thousand, nine hundred and ninety nine"` from a character string to a numeric value.

The idea arose from discussions around {cleanepi} and how messy data could potentially have numbers written in words which would need converting to a numeric datatype for downstream analysis.

I have implemented an [experimental](https://www.reconverse.org/lifecycle.html#experimental) package on https://github.com/bahadzie/numberize

This gave me an opportunity to test our [packagetemplate repo](https://github.com/epiverse-trace/packagetemplate) 

@Bisaloo I will be posting some feedback on using it to build a package. Are [Issues](https://github.com/epiverse-trace/packagetemplate/issues) an acceptable way to do this?

Suggestions for a better package name are open. Submissions so far are {numberwise @adamkucharski}, {text2num @nuredinibrahim}, {str2num @bbah74}. If they're wrong, that is my mishearing. Please correct me.

PS. Is there a way to _italicize Discussion Title_ or does markdown only apply to the body.

**Comment by pratikunterwegs** [comment]
Looks good @bahadzie - I think _numberize_ is a good name, but you might want someone with more imagination to weigh in.

Just commenting on the side issue - some markdown formatting does work in titles, such as monospace formatting, but haven't seen it work for italics.

Also a perhaps silly question - what's the limiting factor on the upper limit of the number that can be converted from text to digits? As in, what's the technical limit preventing conversion of, say, "one quadrillion", "one googol"? Not saying you should add these, just curious where R decides enough is enough.

**Reply by bahadzie** [reply]
> Just commenting on the side issue - some markdown formatting does work in titles, such as monospace formatting, but haven't seen it work for italics.

Monospace is possible using backticks e.g. `Monospace` or is it something else?

There is no technical upper limit AFAIK. I capped it at that because I'm thinking most humans will prefer to write e.g.  2375 than spell it out. If we need to support _quadrillion, googol, ..._ and their respective translations, all we need to do is add them to a dataframe.

**Reply by pratikunterwegs** [reply]
Yes, with backticks - has worked on PRs in the past so might work here too

**Reply by bahadzie** [reply]
Thanks. I'll try that next time I'm starting a discussion

**Comment by juan-umana** [comment]
Looks great @bahadzie ! - Very useful for interoperability of epidemiological reports and line lists among countries.

About package name I think that regarding the one you choose, @bbah74 's {str2num} is a great name for the main function. It feels very coherent with other packages usages.

An observation I want to make (not sure if you are aware) is to be careful between "billion" concept in Spanish and English:

Billion [EN] = A thousand million = 1,000,000,000
Billón [SP] = A million million = 1,000,000,000,000

As well as numbers variations (ex. "veintiuno" and "veintiún" in Spanish for 21).

Just let us know how we can help to document these and other false-friends and variations among English, French and Spanish.

**Reply by bahadzie** [reply]
> Just let us know how we can help to document these and other false-friends and variations among English, French and Spanish.

I would really appreciate it if you and the rest of Team Colombia help me test as many edge cases as possible especially in Spanish because that's the language I have the least experience in. Many thanks in advance.

**Reply by bahadzie** [reply]
> As well as numbers variations (ex. "veintiuno" and "veintiún" in Spanish for 21).

Updated to account for this. Check out the [Examples](https://github.com/bahadzie/numberize#example) in README.md

**Comment by bbah74** [comment]
Hi @bahadzie and others I just managed to solve my login issues with github, the 2FA is a pain :(. Very good comments from @pratikunterwegs and @juan-umana. Just to quickly point out that {str2num} is also used in Matlab, be due diligent to make sure they don't sue us as they are proprietary software.
